frontend/src/components/metrics/TokenInfoPanel.tsx

```diff
@@ -0,0 +1,306 @@
+          <span className="font-mono text-xs text-white/80 break-all">
+            {selectedToken.token_address}
+          </span>
+        </div>
+        <button
+          type="button"
+          className="text-xs text-emerald-400 bg-emerald-900/30 ml-2 px-2 py-1 rounded hover:bg-emerald-400/20 focus:outline-none focus:ring-2 focus:ring-emerald-400 flex items-center"
+          onClick={() => {
+            navigator.clipboard.writeText(selectedToken.token_address);
+          }}
+          aria-label="Copy token address"
+        >
+          <Copy className="w-4 h-4" />
+        </button>
+      </div>
+      {/* Links */}
+      {Array.isArray(links) && links.length > 0 && (
+        <div className="flex flex-wrap gap-2 mb-2">
+          {links.map((link, idx) => {
+            let keyStr = String(idx); // fallback
+            if (typeof link === 'object' && link && 'url' in link) {
+              keyStr = String((link as { url: string }).url || idx);
+            } else if (typeof link === 'string') {
+              keyStr = link;
+            }
+            if (typeof link === 'string') {
+              return (
+                <a
+                  key={keyStr}
+                  href={link}
+                  target="_blank"
+                  rel="noopener noreferrer"
+                  className="text-emerald-400 underline text-xs hover:text-emerald-300"
+                >
+                  {link}
+                </a>
+              );
+            }
+            if (typeof link === 'object' && link !== null && 'url' in link) {
+              return (
+                <a
+                  key={keyStr}
+                  href={(link as { url: string }).url}
+                  target="_blank"
+                  rel="noopener noreferrer"
+                  className="text-emerald-400 underline text-xs hover:text-emerald-300"
+                >
+                  {(link as { label?: string; url: string }).label ||
+                    (link as { url: string }).url}
+                </a>
+              );
+            }
+            return null;
+          })}
+        </div>
+      )}
+
+      {/* Chart Toggle Button */}
+      {lpProgramId && (
+        <button
+          type="button"
+          onClick={() => setIsChartExpanded(!isChartExpanded)}
+          className="w-full flex items-center justify-center gap-2 py-2 mb-2 text-emerald-400 hover:text-emerald-300 focus:outline-none focus:ring-2 focus:ring-emerald-400 rounded"
+        >
+          {isChartExpanded ? (
+            <>
+              <ChevronUp className="w-4 h-4" />
+              <span className="text-sm">Hide Chart</span>
+            </>
+          ) : (
+            <>
+              <ChevronDown className="w-4 h-4" />
+              <span className="text-sm">Show Chart</span>
+            </>
+          )}
+        </button>
+      )}
+
+      {/* GeckoTerminal Chart */}
+      {lpProgramId && isChartExpanded && (
```

Comment:
Can we add stuff on the graph? It'd be nice to add a line when the tweets happened, not sure if its possible.